tests/image_test.py

```diff
@@ -738,6 +713,114 @@ def test_dataframe_iterator_class_mode_categorical_multi_label(self, tmpdir):
+        assert np.array_equal(batch_y[0],
+                              np.array(df['output_0'].tolist()[:3]))
+        assert np.array_equal(batch_y[1],
+                              np.array(df['output_1'].tolist()[:3]))
+        # if one of the outputs is a 2D array
+        df['output_1'] = [np.random.uniform(size=(2, 2, 1))
+                          for _ in range(len(df))]
+        df_iterator = image.ImageDataGenerator().flow_from_dataframe(
+            df, y_col=['output_0', 'output_1'], directory=str(tmpdir),
+            batch_size=3, shuffle=False, class_mode='multi_output'
+        )
+        batch_x, batch_y = next(df_iterator)
+        assert isinstance(batch_x, np.ndarray)
+        assert len(batch_x.shape) == 4
+        assert isinstance(batch_y, list)
+        assert len(batch_y) == 2
+        assert np.array_equal(batch_y[0],
+                              np.array(df['output_0'].tolist()[:3]))
+        assert np.array_equal(batch_y[1],
+                              np.array(df['output_1'].tolist()[:3]))
+        # fail if single column
+        with pytest.raises(TypeError):
+            image.ImageDataGenerator().flow_from_dataframe(
+                df, y_col='output_0',
+                directory=str(tmpdir),
+                class_mode='multi_output'
+            )
+
+    def test_dataframe_iterator_class_mode_raw(self, tmpdir):
+        # save the images in the paths
+        filenames = []
+        count = 0
+        for test_images in self.all_test_images:
+            for im in test_images:
+                filename = 'image-{}.png'.format(count)
+                im.save(str(tmpdir / filename))
+                filenames.append(filename)
+                count += 1
+        # case for 1D output
+        df = pd.DataFrame({"filename": filenames}).assign(
+            output_0=np.random.uniform(size=len(filenames)),
+            output_1=np.random.uniform(size=len(filenames))
+        )
+        df_iterator = image.ImageDataGenerator().flow_from_dataframe(
+            df, y_col='output_0', directory=str(tmpdir),
+            batch_size=3, shuffle=False, class_mode='raw'
+        )
+        batch_x, batch_y = next(df_iterator)
+        assert isinstance(batch_x, np.ndarray)
+        assert len(batch_x.shape) == 4
+        assert isinstance(batch_y, np.ndarray)
+        assert batch_y.shape == (3,)
+        assert np.array_equal(batch_y, df['output_0'].values[:3])
+        # case with a 2D output
+        df_iterator = image.ImageDataGenerator().flow_from_dataframe(
+            df, y_col=['output_0', 'output_1'], directory=str(tmpdir),
+            batch_size=3, shuffle=False, class_mode='raw'
+        )
+        batch_x, batch_y = next(df_iterator)
+        assert isinstance(batch_x, np.ndarray)
+        assert len(batch_x.shape) == 4
+        assert isinstance(batch_y, np.ndarray)
+        assert batch_y.shape == (3, 2)
+        print(batch_y)
+        print(df[['output_0', 'output_0']].values[:3])
```

Comment:
Is printing them(819-820) necessary? I don't think why is there a need to print them considering 821 asserts them.